azurerm/resource_arm_iothub.go

```diff
@@ -110,6 +112,107 @@ func resourceArmIotHub() *schema.Resource {
 				},
 			},
 
+			"endpoint": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"type": {
+							Type:     schema.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								"AzureIotHub.StorageContainer",
+								"AzureIotHub.ServiceBusQueue",
+								"AzureIotHub.ServiceBusTopic",
+								"AzureIotHub.EventHub",
+							}, false),
+						},
+						"connection_string": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"name": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validateIoTHubEndpointName,
+						},
+						"batch_frequency_in_seconds": {
+							Type:         schema.TypeInt,
+							Optional:     true,
+							Default:      300,
+							ValidateFunc: validation.IntBetween(60, 720),
+						},
+						"max_chunk_size_in_bytes": {
+							Type:         schema.TypeInt,
+							Optional:     true,
+							Default:      314572800,
+							ValidateFunc: validation.IntBetween(10485760, 524288000),
+						},
+						"container_name": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"encoding": {
+							Type:             schema.TypeString,
+							Optional:         true,
+							DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(eventhub.Avro),
+								string(eventhub.AvroDeflate),
+							}, true),
+						},
+						"file_name_format": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							ValidateFunc: validateIoTHubFileNameFormat,
+						},
+					},
+				},
+			},
+
+			"route": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringLenBetween(0, 64),
+						},
+						"source": {
+							Type:     schema.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								"DeviceJobLifecycleEvents",
+								"DeviceLifecycleEvents",
+								"DeviceMessages",
+								"Invalid",
+								"TwinChangeEvents",
+							}, false),
+						},
+						"condition": {
+							// The condition is a string value representing device-to-cloud message routes query expression
+							// https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-query-language#device-to-cloud-message-routes-query-expressions
+							Type:     schema.TypeString,
+							Optional: true,
+							Default:  "true",
+						},
+						"endpoint_names": {
+							Type: schema.TypeList,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+							Required: true,
+						},
+						"is_enabled": {
```

Comment:
I'd suggest this could better match other resources in Terraform by being `enabled` rather than `is_enabled` - but this isn't a blocker; what do you think?

Reply:
Good point 👍